### ISSUE TITLE
Add smoke test for `kore-rpc-booster`

### ISSUE THREAD

**Issue by tothtamas28** [runtimeverification/k#4012]
To avoid crashes due to diverging dependency versions between the LLVM and the Booster backend on Nix:

```
 kevm-test> user error (dlopen: /nix/store/9v5d40jyvmwgnq1nj8f19ji2rcc5dksd-glibc-2.37-45/lib/libc.so.6: version `GLIBC_2.38' not found (required by tests/specs/benchmarks/verification/haskell/llvm-library/interpreter.so))
```

**Comment by Baltoli** [comment]
@jberthold we discussed this previously I think?

**Comment by jberthold** [comment]
> @jberthold we discussed this previously I think?

Yes, we merged #3936  which compiles a toy definition with both LLVM and Haskell backends and sends a single (failing) request. I would have thought the glibc issue will show when doing that but maybe
* the environment in which this smoke test is executed is different from the normal user environment, or else
* the test is not executed for some reason (it was a bit confusing to determine when the `smoke` suite would be)

**Comment by jberthold** [comment]
Took a look at recent CI runs, and found
* the `test/smoke/krpc-test` is run in job `K Tests`, but that will run it with natively-built executables under the host system, not under nix.
* For the `Nix flake` jobs, we skip testing unless we are on Mac M1 (where we don't seem to have that glibc problem) https://github.com/runtimeverification/k/blob/20f9eb0ea803fe4bfa4a005c580d22b2a05117e7/.github/workflows/test-pr.yml#L207-L214
* On the Mac M1, the logs do not have any trace of running the `test/smoke` tests either.
* The `Smoke test K` step of the `Nix flake` job does not show any sign of  actually _running_ tests at all in its logs either.
